### PR TITLE
docs(di): add scope section

### PR DESCRIPTION
**Pull request by ukaoskid**
If applied, this PR will add:

- DI docs update with a scope section in Containers.

### CODE REVIEW

File: _docs/di/container.md [hunk 1]
```diff
@@ -76,6 +76,77 @@ DI.set(NewProvider, 'NewProvider');
...
+### Scope
+
+It is possible to manage dependencies within different scopes, and so within different [design patterns](https://en.wikipedia.org/wiki/Design_Patterns).
+
+| Scope             | Description                                                                                                                                    |
+|-------------------|------------------------------------------------------------------------------------------------------------------------------------------------|
+| `default`         | The dependency is a `singleton` and it will be resolved as so without any setting while registering a dependency or injecting via `@Inject()`. |
+| `non-singleton`   | The dependency will be resolved as a brand new instance.                                                                                       |
+
+### Register
```
Comment (alesmit) [inline _docs/di/container.md]: We already have a `### Register` paragraph above, maybe we could review the structure and use a four hashes heading with another title here.

File: _docs/di/container.md [hunk 2]
```diff
@@ -76,6 +76,77 @@ DI.set(NewProvider, 'NewProvider');
...
+  
+  test() {
+    return 'hotdog';
+  }
+}
+
+const DI = new Container();
+
+DI.set(Foo, 'Foo');
+DI.set(Bar, 'Bar');
```
Comment (alesmit) [inline _docs/di/container.md]: If I'm not mistaken this example won't work because `Baz` hasn't been `.set` to the container. Can you please check?

**Review by alesmit — CHANGES_REQUESTED** [review]

File: _docs/di/container.md (hunk 1 above)
Comment (ukaoskid) [reply _docs/di/container.md]: But it is contextual of Scope, I don't know it is a real issue.

File: _docs/di/container.md (hunk 2 above)
Comment (ukaoskid) [reply _docs/di/container.md]: Right, right, correct.

File: _docs/di/container.md (hunk 1 above)
Comment (ukaoskid) [reply _docs/di/container.md]: We could call it like `On register`

**Review by alesmit — APPROVED** [review]

**Pull request merged**